freedomofpress/securedrop-client
# Add DownloadJob

## Description

Closes #406

This is part of breaking down a large PR https://github.com/freedomofpress/securedrop-client/pull/409

This and following smaller refactors are a way to help us avoid bugs around file management when we download and decrypt, which is tricky.

## Summary of changes

* Moved `call_api`, `_download`, and `_check_file_integrity` into a new base class called `DownloadJob` from `FileDownloadJob` and `MessageDownloadJob`.  `_decrypt` will also be moved into `DownloadJob` from its derived classes in a followup PR.

* This also introduces new behavior: checking file integrity of everything that gets downloaded. So now message downloads also get checked.

# Test Plan

No need to test in Qubes, but always feel free. This PR doesn't modify how we decrypt or download, and the extra file integrity checks is not qubes-specific.

1. Send messages and files as a source
2. Refresh on the client to see messages downloaded and decrytped. Click on files to trigger a download and decrypt. When you see "Open" next to the filename you know it's decrypted. 
3. [optional] query the db to make sure statuses look correct
